### ISSUE TITLE
Error: Failed to start capture session: Invalid external buffer capabilities

### ISSUE THREAD

**Issue by aldebaranzbradaradjan** [SunshineStream/Sunshine#100]
### Describe the Bug
Hi,
Sunshine was working great the last week, but I have update my distro and now I can't connect from moonlight.
I got an error  : Error: Failed to start capture session: Invalid external buffer capabilities
```

[2022:03:14:22:52:32]: Info: Found [1] outputs
[2022:03:14:22:52:32]: Info: Virtual Desktop: 2560x1440
[2022:03:14:22:52:32]: Info: XrandR: available
[2022:03:14:22:52:32]: Info: -- Output --
[2022:03:14:22:52:32]: Info:   Resolution: 2560x1440
[2022:03:14:22:52:32]: Info:   Offset: 0x0
[2022:03:14:22:52:32]: Error: Couldn't destroy session handle: Unable to cleanup NvFBC
[2022:03:14:22:52:32]: Error: Environment variable WAYLAND_DISPLAY has not been defined
[2022:03:14:22:52:32]: Info: Detecting connected monitors
[2022:03:14:22:52:32]: Info: //////////////////////////////////////////////////////////////////
[2022:03:14:22:52:32]: Info: //                                                              //
[2022:03:14:22:52:32]: Info: //   Testing for available encoders, this may generate errors.  //
[2022:03:14:22:52:32]: Info: //   You can safely ignore those errors.                        //
[2022:03:14:22:52:32]: Info: //                                                              //
[2022:03:14:22:52:32]: Info: //////////////////////////////////////////////////////////////////
[2022:03:14:22:52:32]: Info: Trying encoder [nvenc]
[2022:03:14:22:52:32]: Info: Screencasting with NvFBC
[2022:03:14:22:52:32]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:32]: Info: Color range: [JPEG]
[2022:03:14:22:52:32]: Error: Could not open codec [h264_nvenc]: Function not implemented
[2022:03:14:22:52:32]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:32]: Info: Screencasting with NvFBC
[2022:03:14:22:52:32]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:32]: Info: Color range: [JPEG]
[2022:03:14:22:52:32]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:32]: Info: Screencasting with NvFBC
[2022:03:14:22:52:32]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:32]: Info: Color range: [JPEG]
[2022:03:14:22:52:32]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:32]: Info: Screencasting with NvFBC
[2022:03:14:22:52:32]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:32]: Info: Color range: [JPEG]
[2022:03:14:22:52:32]: Error: Could not open codec [hevc_nvenc]: Function not implemented
[2022:03:14:22:52:32]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:32]: Info: Screencasting with NvFBC
[2022:03:14:22:52:32]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:32]: Info: Color range: [JPEG]
[2022:03:14:22:52:32]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:32]: Info: Screencasting with NvFBC
[2022:03:14:22:52:33]: Info: Color coding [Rec. 709]
[2022:03:14:22:52:33]: Info: Color range: [JPEG]
[2022:03:14:22:52:33]: Error: Could not open codec [h264_nvenc]: Function not implemented
[2022:03:14:22:52:33]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:33]: Info: Screencasting with NvFBC
[2022:03:14:22:52:33]: Info: Color coding [Rec. 709]
[2022:03:14:22:52:33]: Info: Color range: [JPEG]
[2022:03:14:22:52:33]: Error: cuda::cuda_t doesn't support any format other than AV_PIX_FMT_NV12
[2022:03:14:22:52:33]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:33]: Info: Screencasting with NvFBC
[2022:03:14:22:52:33]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:33]: Info: Color range: [JPEG]
[2022:03:14:22:52:33]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:33]: Info: Screencasting with NvFBC
[2022:03:14:22:52:33]: Info: Color coding [Rec. 601]
[2022:03:14:22:52:33]: Info: Color range: [JPEG]
[2022:03:14:22:52:33]: Warning: nvenc: hevc: replacing nalu prefix data
[2022:03:14:22:52:33]: Error: Couldn't destroy session handle: The context is not bound
[2022:03:14:22:52:33]: Info: 
[2022:03:14:22:52:33]: Info: //////////////////////////////////////////////////////////////
[2022:03:14:22:52:33]: Info: //                                                          //
[2022:03:14:22:52:33]: Info: // Ignore any errors mentioned above, they are not relevant //
[2022:03:14:22:52:33]: Info: //                                                          //
[2022:03:14:22:52:33]: Info: //////////////////////////////////////////////////////////////
[2022:03:14:22:52:33]: Info: 
[2022:03:14:22:52:33]: Info: Found encoder nvenc: [h264_nvenc, hevc_nvenc]
[2022:03:14:22:52:33]: Info: Configuration UI available at [https://localhost:47990]
[2022:03:14:22:52:33]: Error: Failed to create client: Daemon not running
[2022:03:14:22:52:36]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:52:36]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:52:39]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:52:42]: Info: /CN=NVIDIA GameStream Client -- verfied
...
[2022:03:14:22:58:57]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:58:59]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:59:02]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:59:02]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:59:02]: Info: CLIENT CONNECTED
[2022:03:14:22:59:03]: Info: Found [1] outputs
[2022:03:14:22:59:03]: Info: Virtual Desktop: 2560x1440
[2022:03:14:22:59:03]: Info: XrandR: available
[2022:03:14:22:59:03]: Info: -- Output --
[2022:03:14:22:59:03]: Info:   Resolution: 2560x1440
[2022:03:14:22:59:03]: Info:   Offset: 0x0
[2022:03:14:22:59:03]: Error: Couldn't destroy session handle: Unable to cleanup NvFBC
[2022:03:14:22:59:03]: Info: Screencasting with NvFBC
[2022:03:14:22:59:03]: Info: Color coding [Rec. 601]
[2022:03:14:22:59:03]: Info: Color range: [MPEG]
[2022:03:14:22:59:03]: Error: Failed to start capture session: Invalid external buffer capabilities
[2022:03:14:22:59:03]: Error: Couldn't destroy session handle: Unable to cleanup NvFBC
[2022:03:14:22:59:03]: Error: Couldn't release NvFBC context from current thread: Unable to cleanup NvFBC
[2022:03:14:22:59:03]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:03:14:22:59:03]: Info: /CN=NVIDIA GameStream Client -- verfied
```
I have tried differents versions of nvidia patch, recompile sunshine, but no one work.
Any idea ?

### Expected Behavior

Sunshine working like the last week :)

### Additional Context

I use Arch, with an GTX 1070 and nvidia patch from here : https://github.com/keylase/nvidia-patch
Previously I was using aur/nvidia-utils-nvlax that seem to do the same, but like Sunshine wasn't working after update I have tried something else to see.

### Sunshine Host Operating System and Version

Archlinux (last update 14/03/2022)

### Architecture

x86_64

### Sunshine Version

aur/sunshine 0.13.0-1

### GPU Type

Nvidia

### GPU Model

GTX 1070

### GPU Driver/Mesa Version

510.54-1

**Comment by KuleRucket** [comment]
It was caused by an update to vulkan packages. I rolled back to a previous btrfs snapshot and updates packages a few at a time to find which one caused it and narrowed it down. Now, I am blocking the update of these packages in /etc/pacman.conf.

# temporary block as it breaks sunshine
IgnorePkg   =  lib32-vulkan-icd-loader  vulkan-headers  vulkan-icd-loader  vulkan-tools

preventing these updating
warning: lib32-vulkan-icd-loader: ignoring package upgrade (1.2.203-1 => 1.3.207-1)
warning: vulkan-headers: ignoring package upgrade (1:1.3.206-1 => 1:1.3.207-1)
warning: vulkan-icd-loader: ignoring package upgrade (1.2.203-1 => 1.3.207-1)
warning: vulkan-tools: ignoring package upgrade (1.2.203-2 => 1.3.207-1)

**Comment by aldebaranzbradaradjan** [comment]
Nice to know, thanks !
But ignore package update don't seems to be a good idea;
Do you think it's a bug in vulkan packages ? Or something to change in sunshine codebase ?

**Comment by KuleRucket** [comment]
No idea what the actual problem is and for now I don't have any time to debug so this is just a temporary solution. It could be a vulkan problem that will get fixed or it could be that sunshine needs to make adjustments for the newer vulkan libraries and people on other distros will encounter the same problem as they move from vulkan 1.2 to 1.3.

**Comment by KuleRucket** [comment]
On the plus side, OBS also seems to be affected:
https://giters.com/keylase/nvidia-patch/issues/511

**Comment by KuleRucket** [comment]
Fedora 36 will be out of Beta at the end of April and the beta release currently uses vulkan 1.3.204. It will be interesting to see if this also becomes a problem for Fedora users.

**Comment by KuleRucket** [comment]
Tried bumping NvFBC.h to NvFBC Linux API version 1.8 to see if it works better with Vulkan 1.3, but it made no difference. I notice in NvFBC.h that is says you have to use Vulkan 1.1 and it doesn't say >=1.1 like the other libraries. This might be something that needs to be fixed by NVidia.

**Comment by KuleRucket** [comment]
This appears to confirmed as a regression in the vulkan loader covered here:
https://github.com/KhronosGroup/Vulkan-Loader/issues/894

**Comment by KuleRucket** [comment]
Issue can be closed.

**Comment by KuleRucket** [comment]
nvidia dev reported in the vulkan thread it will be fixed in the next driver update.

**Comment by KuleRucket** [comment]
And they did indeed fix it in 510.68.02:

_"Fixed an issue where NvFBC was requesting Vulkan 1.0 while using Vulkan 1.1 core features. This caused NvFBC to fail to initialize with Vulkan loader versions 1.3.204 or newer."_
